packages/esm-patient-list-app/src/ui-components/overflow-menu.component.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useEffect, useRef } from 'react';
+
+interface CustomOverflowMenuComponentProps {
+  menuTitle: React.ReactNode;
+}
+
+const CustomOverflowMenuComponent: React.FC<CustomOverflowMenuComponentProps> = ({ menuTitle, children }) => {
+  const [showMenu, toggleShowMenu] = useState(false);
+  const wrapperRef = useRef(null);
+
+  useEffect(() => {
+    /**
+     * Toggle showMenu if clicked on outside of element
+     */
+    function handleClickOutside(event: MouseEvent) {
+      if (wrapperRef.current && !wrapperRef.current.contains(event.target)) {
+        toggleShowMenu(false);
+      }
+    }
+
+    // Bind the event listener
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      // Unbind the event listener on clean up
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [wrapperRef]);
+
+  return (
+    <div
+      data-overflow-menu
+      className="bx--overflow-menu"
+      style={{
+        width: 'auto',
+        height: 'auto',
+      }}
+      ref={wrapperRef}>
+      <button
+        className={`bx--overflow-menu__trigger ${showMenu && 'bx--overflow-menu--open'}`}
+        aria-haspopup="true"
+        aria-expanded={showMenu}
+        id="custom-actions-overflow-menu-trigger"
+        aria-controls="custom-actions-overflow-menu"
+        onClick={() => toggleShowMenu(!showMenu)}
```

Comment:
Put that in a `useCallback` like this:

```js
const toggle = useCallback(() => toggleShowMenu(state => !state), []);
```

Please rename `toggleShowMenu` to `setShowMenu`.